rekall-core/rekall/plugins/filesystems/tsk.py

```diff
@@ -0,0 +1,162 @@
+import hashlib
+
+import pytsk3 as pytsk3
+import sys
+
+from rekall import obj
+from rekall.plugin import TypedProfileCommand, PhysicalASMixin, ProfileCommand
+from rekall.plugins import guess_profile
+
+TYPE_DIR = 0
+TYPE_FILE = 1
+
+
+class FSEntry(object):
+    def __init__(self, tsk_file):
+        self._tsk_file = tsk_file
+
+    @property
+    def type(self):
+        if self._tsk_file.info.meta.type == pytsk3.TSK_FS_META_TYPE_DIR:
+            return TYPE_DIR
+        else:
+            return TYPE_FILE
+
+    @property
+    def name(self):
+        return self._tsk_file.info.name.name
+
+    @property
+    def size(self):
+        return self._tsk_file.info.meta.size
+
+    def __iter__(self):
+        if self.type == TYPE_DIR:
+            for directory_entry in self._tsk_file.as_directory():
+                if directory_entry.info.meta is None:
+                    continue
+                name = directory_entry.info.name.name
+                if name in [".", ".."]:
+                    continue
+                yield FSEntry(directory_entry)
+        else:
+            return
+
+    def __str__(self):
+        if self.size > 0:
+            return self._tsk_file.read_random(0, self.size)
```

Comment:
This feels like a bad idea - you could accidentally print the file and it can be huge... I suggest to make a special method to get the file data or infact use a file like object.